src/Wellcome.Dds/Wellcome.Dds.Common/DdsIdentifier.cs

```diff
@@ -1,102 +1,147 @@
-﻿using System;
-
-namespace Wellcome.Dds.Common
+﻿namespace Wellcome.Dds.Common
 {
     public class DdsIdentifier
     {
-        public const char Underscore = '_';
-        public const char Slash = '/';
-        public static readonly char[] Separators = { Underscore, Slash };
+        private const char Underscore = '_';
+        private const char Slash = '/';
+        private const string BornDigital = "born-digital";
+        private const string Digitised = "digitised";
+        private static readonly char[] Separators = { Underscore, Slash };
         private readonly string[] parts;
         private readonly string value;
 
-        public string this[int index] => parts[index];
+        /// <summary>
+        /// The identifier for the stored digital object of which this ID might be whole or part.
+        /// This is needed to obtain the object from the storage service, or look up a record in the Catalogue API.
+        /// 
+        /// The majority of DdsIdentifiers are the same string as their packageIdentifiers - e.g.,
+        /// b12312312 or PPCRI/A/B/C
+        /// But for multiple manifestations, different identifiers belong to the same package:
+        /// b19974760_10 and b19974760_10_10 have the same PackageIdentifier, b19974760.
+        /// </summary>
+        public string PackageIdentifier { get; }
+        
+        /// <summary>
+        /// A version of the PackageIdentifier that would only be one path element in a URL or file name
+        /// </summary>
+        public string PackageIdentifierPathElementSafe { get; }
+        
+        /// <summary>
+        /// What prefix in the storage service API is required to locate this object's files
+        /// Currently either "digitised" or "born-digital"
+        /// </summary>
+        public string StorageType { get; }
```

Comment:
Candidate for enum? The methods that use it will then take `StorageType storageType` rather than `string storageType`.

Reply:
The value of this string forms the path element for the storage service - e.g., ../born-digital/... 
it's only assigned within this class, and only read to create that path, so I think it's OK like that.